libs/payments/currency/src/lib/currency.manager.spec.ts

```diff
@@ -0,0 +1,70 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+import { Test, TestingModule } from '@nestjs/testing';
+import { faker } from '@faker-js/faker';
+
+import { CurrencyManager } from './currency.manager';
+import {
+  CurrencyCodeInvalidError,
+  CountryCodeInvalidError,
+  CurrencyCountryMismatchError,
+} from './currency.error';
+import { CURRENCIES_TO_COUNTRIES } from './currency.constants';
+
+describe('CurrencyManager', () => {
+  let mockCurrencyManager: CurrencyManager;
+
+  beforeEach(async () => {
+    const module: TestingModule = await Test.createTestingModule({
+      providers: [CurrencyManager],
+    }).compile();
+
+    mockCurrencyManager = module.get<CurrencyManager>(CurrencyManager);
+  });
+
+  describe('assertCurrencyCompatibleWithCountry', () => {
+    const validCountry = faker.helpers.arrayElement(
+      Object.values(CURRENCIES_TO_COUNTRIES.USD)
```

Comment:
nit: I think `CURRENCIES_TO_COUNTRIES.USD` is an array, so the `Object.values()` wrapper might not be needed.

```js
export const CURRENCIES_TO_COUNTRIES = {
  USD: ['US', 'GB', 'NZ', 'MY', 'SG', 'CA', 'AS', 'GU', 'MP', 'PR', 'VI'],
  EUR: ['FR', 'DE'],
};
```

Reply:
Good catch! Updated!